onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -2379,6 +2386,19 @@ def _infer_BiasSplitGelu(self, node):  # noqa: N802
     def _infer_BiasAdd(self, node):  # noqa: N802
         self._propagate_shape_and_type(node)
 
+    def _infer_RotaryEmbedding(self, node):  # noqa: N802
+        if len(node.output) == 1:
+            self._propagate_shape_and_type(node)
+        elif len(node.output) == 2:
```

Comment:
In spec, the op has only one output. Shall we remove these experiment code?

Reply:
The export of `RotaryEmbedding` with `export_modules_as_functions` sometimes produces extra outputs. The converter team suggested fixing these extra outputs during fusion as it would take time to find the root cause.
https://github.com/microsoft/onnxruntime/blob/d7bb72c9a5598605560d0e99fe9afb0fdf8c0512/onnxruntime/python/tools/transformers/fusion_rotary_attention.py#L599-L601
In order for the optimizer to run, symbolic shape inference needs to initially pass with these extra outputs.

Reply:
I see. Then let's keep it.